sstables/sstables.cc

```diff
@@ -1396,6 +1396,24 @@ void sstable::write_filter() {
     write_simple<component_type::Filter>(filter_ref);
 }
 
+void sstable::maybe_discard_large_filter(uint64_t data_file_size) {
+    if (is_system_keyspace(_schema->ks_name())) {
+        // do not discard filters of system tables
+        return;
+    }
+    assert(has_component(component_type::Filter));
+
+    if (filter_memory_size() < data_file_size / 10) {
+        // filter is not so large - no need to discard
+        return;
+    }
```

Comment:
We need to consider the bloom filter size in totality, not for individual sstables. It could be that a few sstables have relatively large bloom filters, but we're still in budget. 

The best place for this is in sstables_manager, which has a view of all sstables and can choose the best can didate to discard its bloom filter.

Reply:
That means we cannot discard at the time of writing. At the time the sstable with the largest filter is written, the total memory might still be below a limit, the limit might be crossed at a later time, when we write an sstable which has a completely fine filter.

Reply:
In theory, we could have a cache of bloom filters with evictable entries and dynamically load bloom filters on demand.  When evicting an oversized bloom filter we can drop it from disk and recalculate it (with proper sizing) from the index if it's needed to be loaded again. If, in between, the sstable is compacted we'd recalculate the bloom filter fro the compaction output sstables anyhow.

Reply:
It would be good to discard at the time of loading too, so the patch can have an effect on systems that are currently affected by it.

Reply:
let's limit the initial approach to something simple that acts as an emergency brake, not a super-duper thing that will never converge and often not be needed.